docs/rules.md

```diff
@@ -139,6 +139,7 @@ Rules are sorted by severity.
 | `ADMIN_INSTALL_ONLY_REQUIRED`                           | error    | The `browser_specific_settings.gecko.admin_install_only` property must be set to "true" in an enterprise add-on                                                                    |
 | `DATA_COLLECTION_PERMISSIONS_PROP_RESERVED`             | error    | The `browser_specific_settings.gecko.data_collection_permissions` property is reserved for future usage and cannot be used in an extension at the moment                           |
 | `MISSING_DATA_COLLECTION_PERMISSIONS`                   | notice   | The `browser_specific_settings.gecko.data_collection_permissions` will be required in the future. Please add this key to the manifest                                              |
+| `NONE_DATA_COLLECTION_IS_EXCLUSIVE`                     | error    | "none" must be specified without any other data collection permissions                                                                                                             |
```

Comment:
nit: not that big of a deal, but I wonder:

- if it may be worth to more explicitly mention that what this rule describes refers only to the required part of the data collection (in case some reader may be confused and think that that's also for optional data collection, where use of "none" as required along others set as optional is actually fine and expected).
- if the opposite wording may be slightly better, e.g. `"none" must NOT be specified with other required data collection permissions`.